### PR TITLE
Add javascript to allow highlighting multiple lines

### PR DESCRIPTION
**Pull request by alvaroaleman**
Mostly intended for https://steps.ci.openshift.org/ci-operator-reference

### CODE REVIEW

File: pkg/webreg/webreg.go [hunk 1]
```diff
@@ -706,6 +706,112 @@ func WebRegHandler(regAgent agents.RegistryAgent, confAgent agents.ConfigAgent)
 	}
 }
 
+// multiLineHighlightJS allows to append -$NUMBER to one of the line anchors
+// (e.G. `#line18-20`) and will then highlight all the lines.
+const multiLineHighlightJS = `<script>
+function getLinesToHighlight() {
+  let result = [];
+  url = window.location.href;
+  anchorSplitUrl = url.split("#");
```
Comment (xrstf) [inline pkg/webreg/webreg.go:715]: This creates global variables by accident, did you miss `let`?

File: pkg/webreg/webreg.go [hunk 2]
```diff
@@ -706,6 +706,112 @@ func WebRegHandler(regAgent agents.RegistryAgent, confAgent agents.ConfigAgent)
...
+  // No results
+  if (anchorSplitUrl.length < 2) {
+    return result
+  };
```
Comment (xrstf) [inline pkg/webreg/webreg.go:719]: No need for a semicolon here, but one after the return statement above would be nice.

File: pkg/webreg/webreg.go [hunk 3]
```diff
@@ -706,6 +706,112 @@ func WebRegHandler(regAgent agents.RegistryAgent, confAgent agents.ConfigAgent)
...
+    element.style.color = color;
+    element.style.backgroundColor = backgroundColor;
+  });
+
+  window["localStorage"]["highlightLines"] = toHighlight.toString();
+  location.replace(originalURL);
+
+  // Sleep one second, then reset the state store
+  await sleep(1000);
+  window["localStorage"]["highlightLines"] = null;
```
Comment (xrstf) [inline pkg/webreg/webreg.go:807]: How about

```
window.setTimeout(function() {
   window.localStorage.highlightLines = null;
}, 1000);
```

File: pkg/webreg/webreg.go [hunk 4]
```diff
@@ -706,6 +706,112 @@ func WebRegHandler(regAgent agents.RegistryAgent, confAgent agents.ConfigAgent)
...
+}
+
+highlightLines();
+</script>
+
+<body onhashchange="highlightLines()">`
```
Comment (xrstf) [inline pkg/webreg/webreg.go:813]: Instead of putting it on body tag, this can be done in JS via `window.onhashchange = highlightLines;`.

Comment (alvaroaleman) [reply pkg/webreg/webreg.go]: done

File: pkg/webreg/webreg.go (hunk 3 above)
Comment (alvaroaleman) [reply pkg/webreg/webreg.go]: done

File: pkg/webreg/webreg.go (hunk 2 above)
Comment (alvaroaleman) [reply pkg/webreg/webreg.go]: done

File: pkg/webreg/webreg.go (hunk 1 above)
Comment (alvaroaleman) [reply pkg/webreg/webreg.go]: done

**Review by petr-muller — APPROVED** [review]

**Pull request merged**